libs/oracledb/langchain_oracledb/vectorstores/oraclevs.py

```diff
@@ -0,0 +1,1929 @@
+            embedding = self.embedding_function.embed_query(query)
+        documents = self.similarity_search_by_vector(
+            embedding=embedding, k=k, filter=filter, **kwargs
+        )
+        return documents
+
+    async def asimilarity_search(
+        self,
+        query: str,
+        k: int = 4,
+        filter: Optional[Dict[str, Any]] = None,
+        **kwargs: Any,
+    ) -> List[Document]:
+        """Return docs most similar to query."""
+        embedding: List[float] = []
+        if isinstance(self.embedding_function, Embeddings):
+            embedding = await self.embedding_function.aembed_query(query)
+        documents = await self.asimilarity_search_by_vector(
+            embedding=embedding, k=k, filter=filter, **kwargs
+        )
+        return documents
+
+    def similarity_search_by_vector(
+        self,
+        embedding: List[float],
+        k: int = 4,
+        filter: Optional[dict[str, Any]] = None,
+        **kwargs: Any,
+    ) -> List[Document]:
+        docs_and_scores = self.similarity_search_by_vector_with_relevance_scores(
+            embedding=embedding, k=k, filter=filter, **kwargs
+        )
+        return [doc for doc, _ in docs_and_scores]
+
+    async def asimilarity_search_by_vector(
+        self,
+        embedding: List[float],
+        k: int = 4,
+        filter: Optional[dict[str, Any]] = None,
+        **kwargs: Any,
+    ) -> List[Document]:
+        docs_and_scores = await self.asimilarity_search_by_vector_with_relevance_scores(
+            embedding=embedding, k=k, filter=filter, **kwargs
+        )
+        return [doc for doc, _ in docs_and_scores]
+
+    def similarity_search_with_score(
+        self,
+        query: str,
+        k: int = 4,
+        filter: Optional[dict[str, Any]] = None,
+        **kwargs: Any,
+    ) -> List[Tuple[Document, float]]:
+        """Return docs most similar to query."""
+        embedding: List[float] = []
+        if isinstance(self.embedding_function, Embeddings):
+            embedding = self.embedding_function.embed_query(query)
+        docs_and_scores = self.similarity_search_by_vector_with_relevance_scores(
+            embedding=embedding, k=k, filter=filter, **kwargs
+        )
+        return docs_and_scores
+
+    async def asimilarity_search_with_score(
+        self,
+        query: str,
+        k: int = 4,
+        filter: Optional[dict[str, Any]] = None,
+        **kwargs: Any,
+    ) -> List[Tuple[Document, float]]:
+        """Return docs most similar to query."""
+        embedding: List[float] = []
+        if isinstance(self.embedding_function, Embeddings):
+            embedding = await self.embedding_function.aembed_query(query)
+        docs_and_scores = await self.asimilarity_search_by_vector_with_relevance_scores(
+            embedding=embedding, k=k, filter=filter, **kwargs
+        )
+        return docs_and_scores
+
+    @_handle_exceptions
+    def _get_clob_value(self, result: Any) -> str:
```

Comment:
Is this function needed? 
If you use output type handler as i suggested in one of the comment and pass in execute , we get string value from clob directly.